### PR TITLE
Cas 8681

### PR DESCRIPTION
**Pull request by dmehring**
* Create and cache private data structure in MSMetaData::getScanKeys() if the object has been configured to indicate that it is likely to be needed later
* tabs -> spaces in tMSMetaData
* Add filter() method in MSScanKeys needed elsewhere in CASA
* expose MSMetaData::getScanToStatesMap() in public interface for use elsewhere in CASA.

### CODE REVIEW

File: ms/MSOper/MSKeys.cc [hunk 1]
```diff
@@ -161,6 +161,24 @@ std::set<ArrayKey> uniqueArrayKeys(const std::set<ScanKey>& scanKeys) {
     return arrayKeys;
 }
 
+std::set<ScanKey> filter(const std::set<ScanKey> scans, const ArrayKey& arrayKey) {
+    std::set<ScanKey> subset;
+    std::set<ScanKey>::const_iterator iter = scans.begin();
+    std::set<ScanKey>::const_iterator end = scans.end();
+    ArrayKey arrayFromScan;
+    for (; iter!=end; ++iter) {
```
Comment (juliantaylor) [inline ms/MSOper/MSKeys.cc]: this is just std::set_intersection

**Comment by juliantaylor** [comment]
commit 1 and 6 can be squashed:

    git rebase -i cd297d5^

then copy the c762fd2 above cd297d5 and change `pick` to `s`, save, edit the commit message appropriately in the next editor that opens. To update the PR, use `git push -f <yourfork> CAS-8681`

**Comment by dmehring** [comment]
I may have done this incorrectly, because I'm getting conflicts. It might be more efficient for me to start over on another branch, since I don't have a good understanding of what these commands do.

0857 dmehring@legolas:$ git rebase -i cd297d5^
Automatic cherry-pick failed.  After resolving the conflicts,
mark the corrected paths with 'git add <paths>', and
run 'git rebase --continue'
Could not apply c762fd2... Implemented generateScanPropsIfWanted() for reuse

File: ms/MSOper/MSKeys.cc (hunk 1 above)
Comment (dmehring) [inline ms/MSOper/MSKeys.cc]: I'm not understanding how. The types are different, ScanKey vs ArrayKey. How would I turn this into a set_intersection() equivalent?

Comment (juliantaylor) [inline ms/MSOper/MSKeys.cc]: the function can take different typed sets, but you need a ScanKey vs ArrayKey comparison function
if that makes no sense outside of this function its probably not worthwhile to add, then this is fine

**Comment by juliantaylor** [comment]
I meant below cd297d5 not above, sorry

**Pull request closed without merging**